### ISSUE TITLE
Keydown event on input gets cancelled

### ISSUE THREAD

**Issue by neerajbharti** [aspnet/Blazor#803]
Created a blazor project from default template. Then modifed the index.cshtml as the below code. Then type something in the input box. The browser console prints the pressed key, but the keyboard event seems cancelled. Nothing gets printed in the input field.
Also the `arg.Key` property gives me a string. Can I get a integer specifying as to which key was pressed (27 for escape key for example)

```
@page "/"
<input onkeydown="@HandleKeyDown" />
@functions{
    void HandleKeyDown(UIKeyboardEventArgs arg) {
        Console.WriteLine(arg.Key);
    }
}

**Comment by SteveSandersonMS** [comment]
Thanks for logging this, @neerajbharti. We'll make sure to add `arg.KeyCode` or similar, and not to prevent the default action for the `keydown` event. Added to backlog.

> We should have a way to say that event was fully handled in our code or not

Unfortunately that might not be possible in general. The only way it could be possible is if the event raising code is synchronous. But longer term one of our goals is to support running Blazor across a boundard such as in a WebWorker. In that case, all JS-->.NET calls have to be async (even if the .NET->JS ones can be synchronous).

**Comment by Andrzej-W** [comment]
Let's assume I have an input field where I want to enter date in ISO format (YYYY-MM-DD). I want to attach my code to keyboard events and handle `Plus` and `Minus` keys to increment or decrement date. Those keys are fully handled in my code - I don't want to see something like this: `2018-05-12+++` in my input box after a few key press events. Will it be possible or not to implement similar functionality after the bug will be fixed?

**Comment by SteveSandersonMS** [comment]
@Andrzej-W Ultimately it's always possible to implement low-level things like a special keyboard event override in JS and encapsulate that in a component. Hopefully it's also possible to do nicely purely within the Blazor .NET model but would have to try it to determine any rough spots.

**Comment by Andrzej-W** [comment]
Make sure that after the fix this will still work
```
@page "/"

<div onclick="@ClickInDiv" onkeypress="@KeyPressInDiv" style="background-color:aqua;width:300px;height:100px">
    <input onclick="@ClickInInput" onkeypress="@KeyPressInInput" type="text"/>
</div>

@functions{
    int divClicks;
    int inputClicks;

    void ClickInDiv() => Console.WriteLine($"Click in div: { ++divClicks }");
    void ClickInInput() => Console.WriteLine($"Click in input: { ++inputClicks }");
    void KeyPressInDiv(UIKeyboardEventArgs ev) => Console.WriteLine($"KeyPress in div: { ev.Key }");
    void KeyPressInInput(UIKeyboardEventArgs ev) => Console.WriteLine($"KeyPress in input: { ev.Key }");
}
```
Currently it works as expected - I can handle event first in child component (HTML element) and then in parent component.

**Comment by neerajbharti** [comment]
Will we have this fixed in 0.4?

**Comment by danroth27** [comment]
@neerajbharti This isn't currently slated for 0.4. We'll look at it in a future release.
